src/main/java/edu/ie3/datamodel/io/source/csv/CsvDataSource.java

```diff
@@ -364,16 +354,8 @@ protected List<Map<String, String>> csvRowFieldValueMapping(
         .toList();
   }
 
-  /**
-   * Checking the uniqueness for a given entity.
-   *
-   * @param entityName name of the entity
-   * @param allRows of the csv file
-   * @param fields list of set of field names
-   * @return a try object
-   */
-  protected Try<Stream<Map<String, String>>, SourceException> checkUniqueness(
-      String entityName, Collection<Map<String, String>> allRows, List<Set<String>> fields) {
+  protected Set<Map<String, String>> checkExactDuplicates(
```

Comment:
This is for filtering out full (1:1) matching lines.

Reply:
I don't think so, to be honest. It's quite costly because you have to calculate hash values of _all_ parameters combined of all rows. And, crucially, if you check for duplicates with your new implementation, that includes duplicate rows as well because they necessarily also have duplicate keys.